app/gen-server/lib/Housekeeper.ts

```diff
@@ -226,6 +239,22 @@ export class Housekeeper {
     }
   }
 
+  public async checkVersionUpdatesExclusively(): Promise<boolean> {
+    const electionKey = await this._electionStore.getElection('checkVersionUpdates',
+                                                              VERSION_CHECK_PERIOD_MS / 2.0);
+    if (!electionKey) {
+      log.info('Skipping checkVersionUpdates since another server is working on it or worked on it recently');
+      return false;
+    }
+    this._electionKey = electionKey;
+    await this.checkVersionUpdates();
+    return true;
+  }
+
+  public async checkVersionUpdates() {
+    await compareWithLatest(this._server);
+  }
```

Comment:
See my comment on renaming the `compareWithLatest` endpoint - but it might be good to have `update` or a similar verb in the names of these functions, since they're not just checking but actively updating the DB.

Reply:
I liked your renaming suggestion, so I used it.